src/App.js

```diff
@@ -15,11 +15,24 @@ class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      auth: new Auth(this.props.history)
+      auth: new Auth(this.props.history),
+      tokenRenewalComplete: false
     };
   }
+
+  componentDidMount() {
+    const { auth } = this.state;
+    if (auth.isAuthenticated()) {
+      auth.renewToken(() => this.setState({ tokenRenewalComplete: true }));
+    } else {
+      this.setState({ tokenRenewalComplete: true });
+    }
+  }
+
   render() {
     const { auth } = this.state;
+    // Supress rendering until the token renewal check is completed.
+    if (!this.state.tokenRenewalComplete) return "Loading...";
```

Comment:
In a scenario a little bit more complex, I have created a `SplashScreen` HOC that triggers `checkSession` (and after that that signs in on Firebase with a custom token, but that is not important) and that keeps showing a loading screen.

https://github.com/Digituz/krebseng-panel/blob/master/src/components/withSplashScreen.js

Feel free to ping me if you are interested on more info.

Reply:
Nice. Makes sense. Thanks for sharing!